cmd/peer-rest-server.go

```diff
@@ -1137,30 +1137,14 @@ type SpeedtestResult struct {
 	Error     string
 }
 
-// SpeedtestObject implements "random-read" object reader
-type SpeedtestObject struct {
-	buf               []byte
-	remaining         int
-	totalBytesWritten *uint64
+// randomReader returns random data upon every
```

Comment:
Not really.  It returns the same data on every read.

SSD compression may be able to pick that up.

Warp uses a small (random) circular buffer as input to an encryption algorithm. Quite fast and doesn't require a big in-mem buffer and large slow reads from "rand": https://github.com/minio/warp/blob/master/pkg/generator/scambler.go#L84

Reply:
I am using a simpler implementation for now